packages/metro/src/ModuleGraph/worker/__tests__/collectDependencies-test.js

```diff
@@ -42,6 +42,22 @@ const opts = {
   dependencyMapName: null,
 };
 
+it(`collects require context arguments`, () => {
+  const ast = astFromCode(`
+  const a = require.context('./', true, /foobar/);
+`);
+  const {dependencies, dependencyMapName} = collectDependencies(ast, opts);
+
+  expect(dependencies).toEqual([
+    {name: './', data: objectContaining({asyncType: null})},
+  ]);
+  expect(codeFromAst(ast)).toEqual(
+    comparableCode(`
+      const a = require.context(${dependencyMapName}[0], { recursive: true, filter: /foobar/ }, "./");
```

Comment:
I think the code basically needs to be

```suggestion
      const a = require(${dependencyMapName}[0], "./");
```

The way I see it, the values of `recursive` and `filter` will have no effect at runtime, so they only need to be recorded in `dependencies` and not in the transformed code. (You'll want to update the assertion above to check that they are in `dependencies` accordingly.)

This assumes that `${dependencyMapName}[0]` will evaluate at runtime to the ID of an module injected by the bundler, which will have the matching files (after all expansion+filtering) as *its* dependencies. `require(${dependencyMapName][0])` will therefore evaluate to the result of evaluating that module factory - i.e. some object that gives access to those dependencies, as dictated by the `require.context` API.

Reply:
I figured there could be some sort of debugging benefit